packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts

```diff
@@ -491,4 +495,72 @@ describe("middlewareModule", () => {
 
     expect(res).toEqual({ id: 1, name: "Custom method" });
   });
+
+  it("should throw an error as an instance of SdkHttpError class with default http client", async () => {
+    expect.assertions(6);
+    const sdk = initSDK({
+      commerce: buildModule(middlewareModule<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+      }),
+    });
+
+    try {
+      // This is a real request to the middleware, invalid endpoint throws { statusCode: 401, message: "Unauthorized" }
+      await sdk.commerce.unauthorized();
+    } catch (err) {
+      expect(err).toBeInstanceOf(Error);
+      expect(isCausedBySdkHttpError(err)).toBe(true);
+      expect(isSpecificSdkHttpError(err, { statusCode: 401 })).toBe(true);
+      expect(
+        isSpecificSdkHttpError(err, {
+          statusCode: (statusCode) => statusCode === 401,
+        })
+      ).toBe(true);
+      expect(isSdkRequestError(err)).toBe(true);
+      expect(isSdkUnauthorizedError(err)).toBe(true);
+    }
+  });
+
+  it("should throw an error as an instance of SdkHttpError class with axios as a custom http client", async () => {
+    expect.assertions(6);
+
+    const sdk = initSDK({
+      commerce: buildModule(middlewareModule<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+        httpClient: async (url, params, config) => {
```

Comment:
I'm also thinking, it looks like it's `fetch`, not a client. Client suggests that it's an instance of something, instead, it's a fetcher.